cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -820,6 +844,30 @@ protected Future<Boolean> delete(Reconciliation reconciliation) {
                 .map(Boolean.FALSE); // Return FALSE since other resources are still deleted by garbage collection
     }
 
+    /**
+     * Remove the metrics specific to the kind implementing it.
+     *
+     * @param desiredNames  Set of resources which should be reconciled
+     * @param namespace     The namespace to reconcile, or {@code *} to reconcile across all namespaces.
+     */
+    @Override
+    public void removeMetrics(Set<NamespaceAndName> desiredNames, String namespace) {
```

Comment:
Will do

Reply:
Added a whole new test `KafkaAssemblyOperatorMetricsHolderTest` for `KafkaAssemblyOperatorMetricsHolder` to test `removeMetricsForCertificates` as I thought that was more appropiate

Reply:
I don't think that really tests this method. testing this method as part of KAfkaAssemblyOperator might be hard. It might make sense to maybe move it into the MetricsHolder class to make it easier to test? Or have some static method you can test more easily. In any case, I would love to see this logic tested for the different cases that can happen.